src/modules/olMap/components/BaOverlay.js

```diff
@@ -63,83 +116,101 @@ export class BaOverlay extends BaElement {
 		`;
 	}
 
-	/**
-	 * @protected
-	 */
-	_updatePosition() {
-		this._position = this.geometry.getLastCoordinate();
-	}
+	_getContent(model) {
+		const { geometry, overlayType, value } = model;
+		const getStaticDistance = () => {
+			const distance = this._getMeasuredLength(geometry) * value;
+			return this.#unitsService.formatDistance(round(Math.round(distance), -1), 0);
+		};
 
-	/**
-	 * Returns the displayable content of Overlay
-	 * @protected
-	 * @abstract
-	 * @param {string|BaOverlayTypes} type the BaOverlayType
-	 * @returns {string}
-	 */
-	_getContent(/*eslint-disable no-unused-vars */ type) {
-		return this._value;
+		const getDistance = () => {
+			if (canShowAzimuthCircle(geometry)) {
+				// canShowAzimuthCircle() secures that getAzimuth() always returns a valid value except NULL
+				const azimuthValue = getAzimuth(geometry).toFixed(2);
+				const distanceValue = this.#unitsService.formatDistance(this._getMeasuredLength(geometry), 2);
+				return `${azimuthValue}°/${distanceValue}`;
+			}
+			return geometry ? this.#unitsService.formatDistance(this._getMeasuredLength(geometry), 2) : '';
+		};
+
+		const getArea = () => {
+			if (geometry instanceof Polygon) {
+				return this.#unitsService.formatArea(this.#mapService.calcArea(geometry.getCoordinates()), 2);
+			}
+			return '';
+		};
+		switch (overlayType) {
+			case BaOverlayTypes.AREA:
+				return getArea();
+			case BaOverlayTypes.DISTANCE:
+				return getDistance();
+			case BaOverlayTypes.DISTANCE_PARTITION:
+				return geometry ? getStaticDistance() : '';
+			case BaOverlayTypes.HELP:
+			case BaOverlayTypes.TEXT:
+			default:
+				return value;
+		}
 	}
 
 	static get tag() {
 		return 'ba-map-overlay';
 	}
 
+	_getMeasuredLength = (geometry) => {
+		const alreadyMeasuredLength = geometry ? geometry.get(PROJECTED_LENGTH_GEOMETRY_PROPERTY) : null;
```

Comment:
`null` branch not covered by a test

Reply:
Removing null-check. In all usecases the geometry-object is already checked.